File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/ActionReceiver.java
package com.nikhilparanjape.radiocontrol.receivers;

import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.widget.Toast;

/**
 * Created by nikhilparanjape on 7/5/17.
 */

public class ActionReceiver extends BroadcastReceiver {
    private static final String PRIVATE_PREF = "prefs";
    public static final int REQUEST_CODE = 21345;

    @Override
    public void onReceive(Context context, Intent intent) {

        //Toast.makeText(context,"recieved",Toast.LENGTH_SHORT).show();

        final SharedPreferences sharedPref = context.getSharedPreferences(PRIVATE_PREF, Context.MODE_PRIVATE);
        final SharedPreferences.Editor editor = sharedPref.edit();

        editor.putInt("isActive",1);
        editor.apply();
        Toast.makeText(context, "Standby Mode disabled",
                Toast.LENGTH_LONG).show();
        //This is used to close the notification tray
        Intent it = new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS);
        context.sendBroadcast(it);
        NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
        notificationManager.cancel(10110);
    }


}

File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/ActionReceiver.kt
package com.nikhilparanjape.radiocontrol.receivers

import android.annotation.SuppressLint
import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.widget.Toast

/**
 * Created by nikhilparanjape on 7/5/17.
 *
 * @author Nikhil Paranjape
 *
 *
 */

class ActionReceiver : BroadcastReceiver() {

    @SuppressLint("MissingPermission") //Code will only run on devices running < Android 12
    override fun onReceive(context: Context, intent: Intent) {

        //Toast.makeText(context,"received",Toast.LENGTH_SHORT).show();

        val getPrefs = androidx.preference.PreferenceManager.getDefaultSharedPreferences(context)
        val editor = getPrefs.edit()

        editor.putInt("isActive", 1)
        editor.apply()
        Toast.makeText(context, "Standby Mode disabled",
                Toast.LENGTH_LONG).show()
        if(Build.VERSION.SDK_INT < Build.VERSION_CODES.S){
            //This is used to close the notification tray
            val it = Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS)
            context.sendBroadcast(it)
        }

        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        notificationManager.cancel(10110)
    }

    companion object {
        /*private const val PRIVATE_PREF = "prefs"*/
        const val REQUEST_CODE = 21383
    }


}